state/application.go

```diff
@@ -640,29 +654,107 @@ func (a *Application) IsExposed() bool {
 	return a.doc.Exposed
 }
 
+// ExposedEndpoints returns a list of explicitly selected endpoints for this
+// application that should be reachable once the application is exposed. An
+// empty returned list means that all endpoints should be reachable.
+func (a *Application) ExposedEndpoints() []string {
+	if len(a.doc.ExposedEndpoints) == 0 {
```

Comment:
@manadart I have also added these guard blocks to avoid breaking errors that check for a nil `[]string` and not for a `[]string(nil)`. As an alternative, I could patch `setExposed` to check for nil args and generate `$unset` commands but I think it's better if we see the fields in the application docs even if they are empty.